### PR TITLE
Backend api/feature

### PR DESCRIPTION
**Pull request by MaxDaniLOVE**
### Implemented:
- Full integrated API;
- Refactor withModal hoc;
- Rewrite adding tracks;
- Work on styles;
- [Demo](https://dims-beta.netlify.app/)
logins (ADMIN: admin@gmail.com, MENTOR: mentor@gmail.com, USER: mahseeen@gmail.com)
pass: 12345678

### CODE REVIEW

File: src/components/AboutTabs/AboutAuthor.js [hunk 1]
```diff
@@ -1,10 +1,28 @@
...
+        <GithubIcon />
+      </a>
+      <a
+        id='github-link'
+        href='https://www.linkedin.com/in/maksim-danilau-js/'
+        target='_blank'
+        rel='noopener noreferrer'
+      >
+        <LinkedInIcon />
+      </a>
```
Comment (maximsan) [inline src/components/AboutTabs/AboutAuthor.js:24]: Custom Link?

File: src/UI/Buttons/AddProgressButton.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { ButtonDropdown, DropdownToggle, DropdownMenu, DropdownItem } from 'reactstrap';
+
+const AddProgressButton = ({ userTasks, onAddSubtaskModalOpen }) => {
+  const [isOpen, setOpen] = useState(false);
+
+  const toggle = () => setOpen(!isOpen);
```
Comment (maximsan) [inline src/UI/Buttons/AddProgressButton.js:9]: `useCalback` and better wrap in brackets

Comment (MaxDaniLOVE) [reply src/UI/Buttons/AddProgressButton.js:9]: Done

File: src/components/AboutTabs/AboutAuthor.js (hunk 1 above)
Comment (MaxDaniLOVE) [reply src/components/AboutTabs/AboutAuthor.js:24]: Done

**Pull request merged**